### PR TITLE
feat: add allowed values

### PR DESCRIPTION
**Pull request by sfc-gh-kmaurya**
Added allowed_values attribute to snowflake_tag resource

## Test Plan

* These changes has been tested for unit, internal and acceptance tests. 

## References

issue link: https://github.com/chanzuckerberg/terraform-provider-snowflake/issues/815

### CODE REVIEW

File: pkg/resources/tag_test.go [hunk 1]
```diff
@@ -23,17 +23,18 @@ func TestTagCreate(t *testing.T) {
...
+		"comment":        "great comment",
+		"allowed_values": []interface{}{"marketing", "finance"},
 	}
 	d := schema.TestResourceDataRaw(t, resources.Tag().Schema, in)
 	r.NotNil(d)
 
 	WithMockDb(t, func(db *sql.DB, mock sqlmock.Sqlmock) {
 		mock.ExpectExec(
-			`^CREATE TAG "test_db"."test_schema"."good_name" COMMENT = 'great comment'$`,
+			`^CREATE TAG "test_db"."test_schema"."good_name" ALLOWED_VALUES 'marketing', 'finance' COMMENT = 'great comment'$`,
```
Comment (sfc-gh-afedorov) [inline pkg/resources/tag_test.go:37]: add test case like ```CREATE OR REPLACE TAG test."foo[bar]" ALLOWED_VALUES '1', '2["3"]';```

File: pkg/snowflake/tag.go [hunk 2]
```diff
@@ -53,6 +54,22 @@ func (tb *TagBuilder) WithSchema(schema string) *TagBuilder {
...
+// WithAllowedValues adds the allowed values to the query
+func (tb *TagBuilder) WithAllowedValues(av []string) *TagBuilder {
+	tb.allowedValues = AvListToString(av)
+	return tb
+}
+
+func AvListToString(avs []string) string {
+	newavs := avs
+	for index, element := range avs {
+		newavs[index] = fmt.Sprintf(`'%v'`, strings.ReplaceAll(element, "'", "\\'"))
```
Comment (sfc-gh-afedorov) [inline pkg/snowflake/tag.go:66]: can we create or re-use a helper here instead of relying on what `Sprintf` does?

File: pkg/resources/tag.go [hunk 3]
```diff
@@ -287,9 +308,48 @@ func UpdateTag(d *schema.ResourceData, meta interface{}) error {
 		}
 	}
 
+	// If there is change in allowed_values field
+	if d.HasChange("allowed_values") {
+		if _, ok := d.GetOk("allowed_values"); ok {
+			_, n := d.GetChange("allowed_values")
```
Comment (sfc-gh-swinkler) [inline pkg/resources/tag.go:314]: _ , v is more common than using _, n, especially if there is no apparent reason to use "n"

File: pkg/resources/tag.go [hunk 4]
```diff
@@ -287,9 +308,48 @@ func UpdateTag(d *schema.ResourceData, meta interface{}) error {
...
+
+			ns := AvChangeToSliceOfString(n)
```
Comment (sfc-gh-swinkler) [inline pkg/resources/tag.go:316]: this is a funny name for a function. can we call this expandAllowedValues()?

File: pkg/resources/tag.go [hunk 5]
```diff
@@ -287,9 +308,48 @@ func UpdateTag(d *schema.ResourceData, meta interface{}) error {
...
+
+			q := builder.RemoveAllowedValues()
+			err1 := snowflake.Exec(db, q)
```
Comment (sfc-gh-swinkler) [inline pkg/resources/tag.go:319]: this should not be called err1. err is fine

File: pkg/resources/tag.go [hunk 6]
```diff
@@ -287,9 +308,48 @@ func UpdateTag(d *schema.ResourceData, meta interface{}) error {
...
+			if err1 != nil {
+				return errors.Wrapf(err, "error removing ALLOWED_VALUES for tag %v", tag)
+			}
+
+			newValue := ns
```
Comment (sfc-gh-swinkler) [inline pkg/resources/tag.go:324]: this declaration is not needed.

File: pkg/resources/tag.go [hunk 7]
```diff
@@ -287,9 +308,48 @@ func UpdateTag(d *schema.ResourceData, meta interface{}) error {
...
+			addQuery := builder.AddAllowedValues(newValue)
+			err := snowflake.Exec(db, addQuery)
+			if err != nil {
+				return errors.Wrapf(err, "error adding ALLOWED_VALUES for tag %v", tag)
+			}
+		} else {
```
Comment (sfc-gh-swinkler) [inline pkg/resources/tag.go:330]: can you run `go fmt`?

File: pkg/snowflake/tag.go (hunk 2 above)
Comment (sfc-gh-swinkler) [reply pkg/snowflake/tag.go:66]: agreed this should be in helpers function

**Review by sfc-gh-swinkler — CHANGES_REQUESTED** [review]

**Review by sfc-gh-swinkler — APPROVED** [review]

**Comment by sfc-gh-swinkler** [comment]
```
{XDTRAZHROA XDTRAZHROA XDTRAZHROA XDTRAZHROA}
    tag_acceptance_test.go:15: Step 1/1 error: After applying this test step, the plan was not empty.
        stdout:
        
        
        An execution plan has been generated and is shown below.
        Resource actions are indicated with the following symbols:
          ~ update in-place
        
        Terraform will perform the following actions:
        
          # snowflake_tag.test will be updated in-place
          ~ resource "snowflake_tag" "test" {
              ~ allowed_values = [
                  - "",
                ]
                comment        = "Terraform acceptance test"
                database       = "XDTRAZHROA"
                id             = "XDTRAZHROA|XDTRAZHROA|XDTRAZHROA"
                name           = "XDTRAZHROA"
                schema         = "XDTRAZHROA"
            }
        
        Plan: 0 to add, 1 to change, 0 to destroy.
--- FAIL: TestAcc_Tag (4.35s)
```

error during integration test. can you please verify this test case passes and then push a commit to fix?

**Review by sfc-gh-swinkler — APPROVED** [review]

**Pull request merged**